### PR TITLE
Install Runtime IREE Python dependencies for AMD GPU

### PR DESCRIPTION
**Pull request by sogartar**

### CODE REVIEW

File: build_tools/install_runtime_iree_python_deps.sh [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+# Install runtime dependencies for the IREE compiler and runtime.
+# Technically we should not need this if we install the IREE python packages
+# that describe their dependencies.
+# This preinstall will at least speedup subsequent runs.
+# If we use the artifacts from IREE's `cmake install` directly then this script
+# is required as we would not be using wheels.
+
+set -euo pipefail
+
+python -m pip install --no-cache-dir numpy PyYAML
```
Comment (ScottTodd) [inline build_tools/install_runtime_iree_python_deps.sh:18]: No need to bake these into a Docker image (in fact, we shouldn't - it makes updating these difficult, and we know that there's a numpy 2.0 release that we're straddling on both sides of in various repos / developer workflows). We can install them during the job.

In IREE, see the `IREE_BUILD_SETUP_PYTHON_VENV` setting (which runs this code: https://github.com/iree-org/iree/blob/f7ca45d5d58898a2bf9b27c4c7ac09b3b2b48f8f/build_tools/cmake/setup_build.sh#L19-L25) and the direct `python3 -m pip install -r runtime/bindings/python/iree/runtime/build_requirements.txt` commands in other jobs

**Review by ScottTodd — CHANGES_REQUESTED** [review]

File: build_tools/install_runtime_iree_python_deps.sh (hunk 1 above)
Comment (sogartar) [reply build_tools/install_runtime_iree_python_deps.sh:18]: Isn't the argument for preinstalling anything in the docker images to speedup the CI jobs? Why not then have a vanilla docker image and install everything in the CI job?

Comment (ScottTodd) [reply build_tools/install_runtime_iree_python_deps.sh:18]: I untangled some of the Python deps already in https://github.com/iree-org/iree/pull/16346. The jobs still need to download the Docker images, which can take about as long as downloading and installing in pip (e.g. if the install doesn't perform other actions like compiling from source).

Any fast changing and easy to install deps can just be installed in the jobs themselves. Slower changing or harder to install deps like GPU drivers and toolchains / compilers can be included in Docker images.

**Comment by sogartar** [comment]
I will move this installs to the CI jobs.

**Pull request closed without merging**